src/server/command_registry.h

```diff
@@ -169,8 +169,17 @@ class CommandRegistry {
   CommandRegistry& operator<<(CommandId cmd);
 
   const CommandId* Find(std::string_view cmd) const {
-    auto it = cmd_map_.find(cmd);
-    return it == cmd_map_.end() ? nullptr : &it->second;
+    if (const auto it = cmd_map_.find(cmd); it != cmd_map_.end()) {
+      return &it->second;
+    }
+
+    if (const auto it = cmd_aliases_.find(cmd); it != cmd_aliases_.end()) {
+      if (it->first == it->second) {
```

Comment:
is this condition to make sure that you you dont call find in a loop?
something like __ping=__ping ?
we can reject this on load i.e exit if key=value

Reply:
We do reject `key=value` on load in https://github.com/dragonflydb/dragonfly/blob/abhijat/feat/allow-cmd-aliasing/src/server/command_registry.cc#L95-L98

I have changed the behavior of `dfly::CommandRegistry::Find` so instead of recursive calls we just lookup once in the alias map and then once in the cmd map (if alias was found). This should take care of cycles. I also removed the `first == second` check as it is no longer required.